jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/WorkspaceConverterMojo.java

```diff
@@ -184,369 +185,374 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
     Map<String, MavenModule> moduleMappings = loadMavenModule(project.getFile().getPath().replace("pom.xml", ""));
     Optional<Package> optionalPackage = getOrCreatePackage(packageRegistry, findPackageNameByReference(String.format("%s:%s", project.getGroupId(), project.getArtifactId()), moduleMappings), null, moduleMappings);
-    try {
-      synchronized (lock) {
-        rootPackageJson.readPackageJson();
-        moduleMappings.entrySet().stream()
-                .map(entry -> {
-                  if (ModuleType.IGNORE == entry.getValue().getModuleType()) {
-                    return null;
-                  } else if (ModuleType.AGGREGATOR == entry.getValue().getModuleType() && !isAggregatorToConvert(entry.getValue().getData())) {
-                    return null;
-                  } else {
-                    return getPackageFolderName(entry.getKey());
-                  }
-                })
-                .filter(Objects::nonNull)
-                .forEach(rootPackageJson::addWorkspace);
-
-        rootPackageJson.writePackageJson();
-
-        File eslintRcFile = Paths.get(convertedWorkspaceTarget + "/.eslintrc.js").toFile();
-        FileUtils.copyInputStreamToFile(getClass().getResourceAsStream("/net/jangaroo/jooc/mvnplugin/.eslintrc.js"), eslintRcFile);
-
-        Map<String, Object> lernaJson = new LinkedHashMap<>();
-        lernaJson.put("useWorkspaces", true);
-        lernaJson.put("version", "0.0.0");
-        Map<String, Object> lernaCommandMap = new LinkedHashMap<>();
-        Map<String, Object> lernaRunMap = new LinkedHashMap<>();
-        lernaRunMap.put("stream", true);
-        lernaCommandMap.put("run", lernaRunMap);
-        lernaJson.put("command", lernaCommandMap);
-
-        FileUtils.write(new File(convertedWorkspaceTarget + "/lerna.json"), objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(lernaJson).concat("\n"));
+    if (!new File(convertedWorkspaceTarget).exists()) {
+      new File(convertedWorkspaceTarget).mkdirs();
+    }
+    synchronized (session.getLocalRepository()) {
```

Comment:
Could you please try one last variant, `session.getRequest()`? I still think using the local repository feels weird, the request would better represent what we mean, namely to synchronize on "the Maven run".
And please add a comment why we do this, something like "a static object is not enough, because Maven plugins are invoked with different class loaders, so we need some object that is the same for the whole Maven run".

Reply:
Removed the synchronize